# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1744150468 -7200
# Parent  359c1515021d69431a9790eeb89e44d68c46fe89
Misc: Use Betterbird's own donation appeal URL.
* * *
Misc: Use own prefs for BB donation appeal.
* * *
Backout hunk that opens appeal in application tab
* * *
Reinstated appeal that TB backed out in https://hg-edge.mozilla.org/comm-central/rev/629eec284a7986ee68a4b996424d04d822367be9

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1411,8 +1411,15 @@ pref("services.sync.engine.identities", 
 pref("services.sync.engine.prefs", false);
 pref("services.sync.engine.servers", true);
 #endif
 
+// Donation appeal.
+// Use our own prefs for the end-of-year appeal. Otherwise running TB on the
+// profile would show the TB appeal and then never again the BB appeal.
+pref("betterbird.donation.eoy.version", 2024);
+pref("betterbird.donation.eoy.version.viewed", 0);
+pref("betterbird.donation.eoy.url", "https://www.betterbird.eu/end-of-year/");
+
 // Unified toolbar
 
 // 0: icons beside text
 // 1: icons above text
diff --git a/mail/base/content/messenger.js b/mail/base/content/messenger.js
--- a/mail/base/content/messenger.js
+++ b/mail/base/content/messenger.js
@@ -311,8 +311,14 @@ var gMailInit = {
       await atStartupRestoreTabs(false);
       gSpacesToolbar.onLoad();
     }
 
+    // Show the end of year donation appeal page.
+    if (this.shouldShowEOYDonationAppeal()) {
+      // Add a timeout to prevent opening the browser immediately at startup.
+      setTimeout(this.showEOYDonationAppeal, 2000);
+    }
+
     if (Services.prefs.getBoolPref("mail.inappnotifications.enabled", false)) {
       import("chrome://messenger/content/in-app-notification-manager.mjs")
         .then(() => {
           document
@@ -350,8 +356,57 @@ var gMailInit = {
     document.getElementById("tabmail")._teardown();
 
     OnMailWindowUnload();
   },
+
+  /**
+   * Check if we can trigger the opening of the donation appeal page.
+   *
+   * @returns {boolean} - True if the donation appeal page should be opened.
+   */
+  shouldShowEOYDonationAppeal() {
+    const currentEOY = Services.prefs.getIntPref(
+      "betterbird.donation.eoy.version",
+      2023
+    );
+    const viewedEOY = Services.prefs.getIntPref(
+      "betterbird.donation.eoy.version.viewed",
+      0
+    );
+
+    // Short-circuit further checking.
+    if (viewedEOY >= currentEOY) {
+      return false;
+    }
+
+    const now = Math.round(Date.now() / 1000);
+    // If we never checked, pretend we checked now.
+    const lastCheckTime = Services.prefs.getIntPref(
+      "betterbird.update.lastCheckTime",
+      now
+    );
+
+    // Display one day after installation the earliest. 24 * 60 * 60.
+    return now - lastCheckTime > 86400 && !Cu.isInAutomation;
+  },
+
+  /**
+   * Open the end of year appeal in a new web browser page. We don't open this
+   * in a tab due to the complexity of the donation site, and we don't want to
+   * handle that inside Thunderbird.
+   */
+  showEOYDonationAppeal() {
+    const url = Services.prefs.getStringPref("betterbird.donation.eoy.url");
+    openLinkExternally(url);
+    const currentEOY = Services.prefs.getIntPref(
+      "betterbird.donation.eoy.version",
+      2023
+    );
+    Services.prefs.setIntPref(
+      "betterbird.donation.eoy.version.viewed",
+      currentEOY
+    );
+  },
 };
 
 /**
  * Called at startup to verify if we have ny existing account, even if invalid,
